File: clojure/acronym/src/acronym.clj
(ns acronym
  (:require [clojure.string :as str]))

(defn insert-space-before-uppercase [s]
  (str/replace s #"(?<=[a-z])(?=[A-Z])" " "))

(defn split-space-and-hyphen [s]
  (str/split s #"[ -]"))

(defn acronym
  "Converts phrase to its acronym."
  [phrase]
  (->> phrase
       (insert-space-before-uppercase)
       (split-space-and-hyphen)
       (map first)
       (map str)
       (map str/upper-case)
       (str/join)))

File: solutions/kotlin/acronym/2/src/main/kotlin/Acronym.kt
object Acronym {
    fun generate(phrase: String): String = buildString {
        var wordStart = true
        for (c in phrase) {
            if (wordStart && (c in 'a'..'z' || c in 'A'..'Z')) {
                append(c)
                wordStart = false
            } else if (!wordStart && (c == ' ' || c == '-')) {
                wordStart = true
            }
        }
    }.uppercase()
}
